torchvision/datasets/kinetics.py

```diff
@@ -20,44 +34,104 @@ class Kinetics400(VisionDataset):
+            - video (Tensor[T, H, W, C]): the `T` video frames in torch.uint8 tensor
             - audio(Tensor[K, L]): the audio frames, where `K` is the number of channels
-              and `L` is the number of points
+              and `L` is the number of points in torch.float tensor
             - label (int): class of the video clip
+
+    Raises:
+        RuntimeError: If ``download is True`` and the video archives are already extracted.
     """
 
-    def __init__(self, root, frames_per_clip, step_between_clips=1, frame_rate=None,
-                 extensions=('avi',), transform=None, _precomputed_metadata=None,
-                 num_workers=1, _video_width=0, _video_height=0,
-                 _video_min_dimension=0, _audio_samples=0, _audio_channels=0):
-        super(Kinetics400, self).__init__(root)
+    _TAR_URLS = {
+        "400": "https://s3.amazonaws.com/kinetics/400/{split}/k400_{split}_path.txt",
+        "600": "https://s3.amazonaws.com/kinetics/600/{split}/k600_{split}_path.txt",
+        "700": "https://s3.amazonaws.com/kinetics/700_2020/{split}/k700_2020_{split}_path.txt",
+    }
+    _ANNOTATION_URLS = {
+        "400": "https://s3.amazonaws.com/kinetics/400/annotations/{split}.csv",
+        "600": "https://s3.amazonaws.com/kinetics/600/annotations/{split}.txt",
+        "700": "https://s3.amazonaws.com/kinetics/700_2020/annotations/{split}.csv",
+    }
+
+    def __init__(
+        self,
+        root: str,
+        frames_per_clip: int,
+        num_classes: str = "400",
+        split: str = "train",
+        frame_rate: float = None,
+        step_between_clips: int = 1,
+        transform: Optional[Callable] = None,
+        extensions=("avi", "mp4"),
+        download: bool = False,
+        num_download_workers=1,
+        num_workers: int = 1,
+        _precomputed_metadata=None,
+        _video_width=0,
+        _video_height=0,
+        _video_min_dimension=0,
+        _audio_samples=0,
+        _audio_channels=0,
+        **kwargs
+    ) -> None:
+
+        # TODO: support test
+        verify_str_arg(split, arg="split", valid_values=['train', 'val', 'unknown'])
+        verify_str_arg(num_classes, arg="num_classes", valid_values=["400", "600", "700"])
+        self.num_classes = num_classes
+        self.extensions = extensions
+        self.num_download_workers = num_download_workers
+
+        _use_legacy_structure = kwargs.get('_use_legacy_structure', False)
+        if _use_legacy_structure:
+            print("Using legacy structure")
+            self.root = root
+            self.split_folder = root
+            self.split = "unknown"
+            assert download == False, "Cannot download the videos using legacy_structure."
+        else:
+            self.root = root
+            self.split_folder = path.join(root, split)
+            self.split = split
+
+        if download:
+            self.download_and_process_videos()
+
+        super().__init__(self.split_folder)
```

Comment:
We should pass `root` here so `root == Kinectics(root, ...).root`. Internally we can then use `split_folder` for everything.